src/Build.UnitTests/TerminalLoggerConfiguration_Tests.cs

```diff
@@ -232,6 +231,30 @@ public void TerminalLoggerDefaultOff(string defaultValue)
         ShouldNotBeTerminalLog(output);
     }
 
+    [WindowsFullFrameworkOnlyTheory]
```

Comment:
Why restrict to Windows/full framework only?

Reply:
I guess it's because the bug doesn't reproduce with `dotnet build`. Not sure why, may be worth investigating.

Reply:
The reason is more trivial:
"The task "RegisterAssembly" is not supported on the .NET Core version of MSBuild. Please use the .NET Framework version of MSBuild."

Reply:
Does it really require the RegisterAssembly task to reproduce though? https://github.com/dotnet/msbuild/pull/9388#discussion_r1380379337